pkg/services/govmomi/cluster/service.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/session"
+)
+
+type computeClusterContext interface {
+	context.Context
+
+	GetSession() *session.Session
+
+	GetLogger() logr.Logger
+}
+
+func ListHostsFromGroup(ctx context.Context, ccr *object.ClusterComputeResource, hostGroup string) ([]object.Reference, error) {
+	clusterConfigInfoEx, err := ccr.Configuration(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var refs []object.Reference
+	for _, group := range clusterConfigInfoEx.Group {
+		if clusterHostGroup, ok := group.(*types.ClusterHostGroup); ok {
+			if clusterHostGroup.Name == hostGroup {
+				for _, managedRef := range clusterHostGroup.Host {
+					refs = append(refs, managedRef)
+				}
+				return refs, nil
+			}
+		}
+	}
+	return refs, nil
+}
+
+func reconfigure(ctx context.Context, ccr *object.ClusterComputeResource, spec types.BaseComputeResourceConfigSpec) error {
+	reconfigureTask, err := ccr.Reconfigure(ctx, spec, true)
+	if err != nil {
+		return err
+	}
+
+	err = reconfigureTask.Wait(ctx)
```

Comment:
we should probably revisit this to stop blocking CAPV's reconciliation, see the `defer reconcileVSphereVMOnTaskCompletion(ctx)` on `ReconcileVM`

Reply:
This reconfigure task is not a long-running task, does it still make sense to defer the reconciliation.